### PR TITLE
libc: Don't define localtime[_r] to macro when CONFIG_LIBC_LOCALTIME not define

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since libc++ declare these function in ctime by:
```
using ::localtime[_r];
```

### CODE REVIEW

File: arch/renesas/src/rx65n/rx65n_rtc.c [hunk 1]
```diff
@@ -774,7 +774,6 @@ int up_rtc_settime(FAR const struct timespec *tp)
 
   /* Day of the week (0-6, 0=Sunday) */
 
-#if defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED)
```
Comment (btashton) [inline arch/renesas/src/rx65n/rx65n_rtc.c:777]: Should this be just conditional on CONFIG_TIME_EXTENDED now?

File: arch/renesas/src/rx65n/rx65n_rtc.c [hunk 2]
```diff
@@ -1355,9 +1353,7 @@ int up_rtc_getdatetime(FAR struct tm *tp)
 
       /* Days since Sunday (0-6) */
 
-#if defined(CONFIG_LIBC_LOCALTIME) || defined(CONFIG_TIME_EXTENDED)
```
Comment (btashton) [inline arch/renesas/src/rx65n/rx65n_rtc.c:1358]: Same question

File: arch/renesas/src/rx65n/rx65n_rtc.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/renesas/src/rx65n/rx65n_rtc.c:777]: The CONFIG_TIME_EXTENDED is redundant after this commit:
```
commit 1e3f457c9ef71cbaa751d8ad8348ad1993b956b0
Author: Xiang Xiao <xiaoxiang@xiaomi.com>
Date:   Thu Jan 23 08:17:22 2020 -0600

    Remove TIME_EXTENDED option to more conform C standard
    
    Gregory Nutt <gnutt@nuttx.org>
    
        Run all .c and .h files modified in this PR through nxstyle and correct all coding standard problems.
    
    Xiang Xiao <xiaoxiang@xiaomi.com>
    
        Remove TIME_EXTENDED option to more conform C standard
    
        Note: the code/data size increment is small
```

Comment (btashton) [reply arch/renesas/src/rx65n/rx65n_rtc.c:777]: Thanks for clarifying.

**Pull request merged**